File: docs/demos/old-api.py
from flask import Flask, jsonify, request
import psycopg2

app = Flask(__name__)

def get_db():
    return psycopg2.connect("dbname=demo user=admin")

@app.route("/users")
def list_users():
    db = get_db()
    cur = db.cursor()
    cur.execute("SELECT id, name, email FROM users")
    rows = cur.fetchall()
    cur.close()
    db.close()
    return jsonify([{"id": r[0], "name": r[1], "email": r[2]} for r in rows])

@app.route("/users/<int:user_id>")
def get_user(user_id):
    db = get_db()
    cur = db.cursor()
    cur.execute("SELECT id, name, email FROM users WHERE id = %s", (user_id,))
    row = cur.fetchone()
    cur.close()
    db.close()
    if row:
        return jsonify({"id": row[0], "name": row[1], "email": row[2]})
    return jsonify({"error": "not found"}), 404

File: docs/demos/old-api.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\JsonResponse;
use Illuminate\Support\Facades\DB;

class UserController extends Controller
{
    public function index(): JsonResponse
    {
        $users = DB::select('SELECT id, name, email FROM users');

        return response()->json($users);
    }

    public function show(int $id): JsonResponse
    {
        $user = DB::select('SELECT id, name, email FROM users WHERE id = ?', [$id]);

        if (empty($user)) {
            return response()->json(['error' => 'not found'], 404);
        }

        return response()->json($user[0]);
    }

    public function store(): JsonResponse
    {
        $name = request('name');
        $email = request('email');

        DB::insert('INSERT INTO users (name, email) VALUES (?, ?)', [$name, $email]);

        return response()->json(['message' => 'created'], 201);
    }
}
